schema_utils.py

```diff
@@ -185,6 +204,223 @@ def get_validator(validator_id):
     return _Validators.get(validator_id)
 
 
+def is_numeric_operand(string):
+    """Checks if the given string represents a valid number.
+
+    Args:
+        string: str. The operand to be validated.
+
+    Returns:
+        bool. Whether the given string is a valid numeric operand.
+    """
+    try:
+        float(string)
+        return True
+    except ValueError:
+        return False
+
+
+def is_algebraic_operand(string):
+    """Checks if the given string represents a valid algebraic operand
+    (variable). Given string must be a singular english alphabet or a valid
+    greek letter.
+
+    Args:
+        string: str. The operand to be validated.
+
+    Returns:
+        bool. Whether the given string is a valid algebraic operand.
+    """
+    return (
+        string.isalpha() and len(string) == 1) or string in GREEK_LETTERS.keys()
+
+
+def is_operand(string):
+    """Checks if the given string represents a valid algebraic/numeric operand.
+
+    Args:
+        string: str. The operand to be validated.
+
+    Returns:
+        bool. Whether the given string is a valid algebraic/numeric operand.
+    """
+    return is_numeric_operand(string) or is_algebraic_operand(string)
+
+
+def contains_balanced_brackets(expression):
+    """Checks if the given expression contains a balanced bracket sequence.
+
+    Args:
+        expression: str. A math expression (algebraic/numeric).
+
+    Returns:
+        bool. Whether the given expression contains a balanced
+            bracket sequence.
+    """
+    openers, closers = '({[', ')}]'
+    stack = []
+    for character in expression:
+        if character in openers:
+            stack.append(character)
+        elif character in closers:
+            if len(stack) == 0:
+                return False
+            top_element = stack.pop()
+            if openers.index(top_element) != closers.index(character):
+                return False
+    return len(stack) == 0
+
+
+def tokenize(expression):
+    """Tokenizes the given math expression into separate components.
+
+    Args:
+        expression: str. A math expression (algebraic/numeric).
+
+    Returns:
+        list. A list of tokens present in the expression.
+    """
+
+    # Removing whitespace.
+    expression = expression.replace(' ', '')
+
+    # Replacing greek letters with corresponding symbols.
+    for letter in GREEK_LETTERS:
+        expression = expression.replace(letter, GREEK_LETTERS[letter])
+
+    # Replacing all parenthesis with () since once the balance is validated, the
+    # type of parenthesis makes no difference for syntactic validation.
+    expression = expression.replace('{', '(')
+    expression = expression.replace('}', ')')
+    expression = expression.replace('[', '(')
+    expression = expression.replace(']', ')')
+
+    tokens = []
+
+    i = 0
+    while i < len(expression):
+        current_token = []
```

Comment:
I've changed the name of the variable.
The only reason I used a list instead of a string is that I'm appending characters to this while iterating through the loop so it's more efficient to append items to a list as compared to a string since strings are immutable.